### ISSUE TITLE
Rule only works in conjunction with Apple Bluetooth Keyboard

### ISSUE THREAD

**Issue by wheredoesyourmindgo** [pqrs-org/Karabiner-Elements#2770]
I have the following simple rule in karabiner.json. If I hold down shift on an Apple Bluetooth keyboard, i can press F24 on a macropad (different device input) and the rule works as expected. Strangely, if I hold down shift on a different keyboard, one other than an Apple Bluetooth keyboard and press F24 on that same macropad the rule doesn't work. Is there a workaround for this? There really isn't anything in the log that eludes to any problems. Karabiner Event Viewer shows the same keycodes firing when I inspect those two scenarios.

Using Karabiner 13.5.0
macOS 11.4

```
          {
            "description": "Make Screen Dimmer",
            "manipulators": [
              {
                "from": {
                  "key_code": "f24",
                  "modifiers": {
                    "mandatory": ["shift"]
                  }
                },
                "to": [
                  {
                    "key_code": "f19"
                  }
                ],
                "type": "basic"
              }
            ]
          },
```

**Comment by MuhammedZakir** [comment]
> [...] Karabiner Event Viewer shows the same keycodes firing when I inspect those two scenarios.

What do you mean by "same"? It it's "same", it should work. Please do this:

1. Create a new profile.
2. Add only the rule you have posted.
3. Open EventViewer
3. Hold down shift on Apple Bluetooth keyboard and press F24 on macropad.
4. Post the EventVeiwer output here.
5. Clear EventViewer output.
6. Hold down shift on the other keyboard and press F24 on macropad.
7. Post the EventViewer ouput here.

**Comment by wheredoesyourmindgo** [comment]
It appears they are not the same (apologies); after following your instruction the output is as follows.

#### Apple Bluetooth Keyboard

|Event Type|HID Usage|Name|Misc|
|---|---|---|---|
|down      |      7,229|     {"key_code":"right_shift"}                       |      flags right_shift|
|up          |    7,229    | {"key_code":"right_shift"}                         |    |
|down    |        7,110   |  {"key_code":"f19"}                                   |  |
|up        |      7,110      |{"key_code":"f19"}                                     ||
|down  |         7,229   |  {"key_code":"right_shift"}                           |  flags right_shift|
|up      |        7,229      |{"key_code":"right_shift"}                             ||

#### Other (Qmk firmware) Keyboard

|Event Type|HID Usage|Name|Misc|
|---|---|---|---|
|down        |    7,229  |   {"key_code":"right_shift"}|                             flags right_shift|
|down       |     7,115   |  {"key_code":"f24"}             |                        |
|up       |       7,115|     {"key_code":"f24"}        |                             |
|up      |        7,229|     {"key_code":"right_shift"}                              ||

**Comment by MuhammedZakir** [comment]
That's weird indeed. This is not something I can help with. Sorry! :-(

**Comment by MuhammedZakir** [comment]
Have you enabled your keyboard in KE preferences?

**Comment by pcwa-ahendricks** [comment]
Yeah, that keyboard works fine (including shift presses). But Karabiner won't orchestrate the two inputs so to speak; Karabiner seems to orchestrate just fine if using the Apple Keyboard.

**Comment by MuhammedZakir** [comment]
> Yeah, that keyboard works fine (including shift presses). But Karabiner won't orchestrate the two inputs so to speak; Karabiner seems to orchestrate just fine if using the Apple Keyboard.

What I meant was, if you enabled your keybaord **in** `Karabiner-Elements Preferences > Devices` [1][2]. Also, test if KE is able to modify that keyboard using a simple/complex modification.

[1] https://karabiner-elements.pqrs.org/docs/manual/configuration/configure-devices/
[2] I don't remember, but I think EventViewer will show keyboard events for disabled keyboards.

**Comment by wheredoesyourmindgo** [comment]
Wow. It's working now. The keyboards were enabled, but it seems that the fix is to also enable the duplicated device with the mouse symbol on it. I've included a screenshot. Thanks @MuhammedZakir; Much appreciated. I'm not sure if this is technically an issue/bug. Anyone who is maintaining this repo please feel free to close this issue since the workaround seems to effectively be a solution for this issue. I'll leave it open just in case it is a bug.

<img width="1112" alt="Screen Shot 2021-07-21 at 6 55 57 PM" src="https://user-images.githubusercontent.com/3316678/126580670-ccd8abae-d9e3-4695-bccf-f57846dddc30.png">

**Comment by MuhammedZakir** [comment]
Mouses are disabled by default. I guess KE should enable that device if it's types are both keyboard and mouse. You may edit the title and post to say that rather than closing this.
